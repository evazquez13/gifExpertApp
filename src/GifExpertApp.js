import React, { useState } from 'react'
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';

const GifExpertApp = ({defaultCategories=[]}) => {

    const [categories, setCategories] = useState(defaultCategories);
    
    return (
        <>            
            <h2>GifExpertApp</h2>
            <AddCategory setCategories={setCategories} />
            <hr/>
            <ol>
                {
                    categories.map(categoria => (
                        <GifGrid 
                            key={categoria}
                            category={categoria}/>
                    ))
                }
            </ol>
        </>
    )
}

export default GifExpertApp;
