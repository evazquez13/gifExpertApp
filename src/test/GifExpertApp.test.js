import React from "react";
import { shallow } from "enzyme";
import GifExpertApp from "../GifExpertApp";

describe('Pruebas al componente <GifExpertApp/>', () =>{

    
    test('Validar que el componente se muestre correctamente', () => {
        
        const wrapper = shallow(<GifExpertApp/>);
        expect(wrapper).toMatchSnapshot();

    });

    test('Debe mostrar una lista de categorias', () => {

        const categorias = ['One punch','Dragon ball'];
        const wrapper = shallow(<GifExpertApp defaultCategories={categorias}/>);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categorias.length);
                
    });
    
    

});