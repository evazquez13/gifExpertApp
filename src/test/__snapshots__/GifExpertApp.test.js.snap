// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas al componente <GifExpertApp/> Debe mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One punch"
      key="One punch"
    />
    <GifGrid
      category="Dragon ball"
      key="Dragon ball"
    />
  </ol>
</Fragment>
`;

exports[`Pruebas al componente <GifExpertApp/> Validar que el componente se muestre correctamente 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
